### ISSUE TITLE
About the implement of maskConv2d

### ISSUE THREAD

**Issue by grimoire** [open-mmlab/mmcv#460]
Hi

I am trying to understand the implement detail of masked conv2d.

```
import mmcv.ops
import torch

conv_param = [3,6,3,1,0]

conv_mask = mmcv.ops.MaskedConv2d(*conv_param).cuda()
conv_torch = torch.nn.Conv2d(*conv_param).cuda()
conv_torch.weight = conv_mask.weight
conv_torch.bias = conv_mask.bias

x = torch.rand(1,3,32,32).cuda()
mask = x[:,0,:,:]>0.5

mask_result = conv_mask(x,mask)
torch_result = conv_torch(x)*mask.float().unsqueeze(1)[:,:,:-2,:-2]

diff = (mask_result-torch_result).abs().max()
```

I expect the diff is zeros... but it give me a relatively large number.
And in https://github.com/open-mmlab/mmcv/blob/master/mmcv/ops/csrc/masked_conv2d_cuda_kernel.cuh MaskedCol2imForward

```
    const int m_index = index % mask_cnt;
    const int h_im = mask_h_idx[m_index];
    const int w_im = mask_w_idx[m_index];
    const int c_im = index / mask_cnt;
    // compute the start and end of the output
    data_im[(c_im * height + h_im) * width + w_im] = data_col[index];
```

the h_im and w_im comes from mask(which have the same height and width with input), but the data_im is a ptr of output(which might have different height and width). What if h_im>=height or w_im>=width, There might be some kind of conflict?

**Comment by hellock** [comment]
Thanks for reporting and this is a potential bug. In the models that adopts MaskedConv2d, input and output sizes are the same, and current implementation does not support a smaller output feature map.
